sdk/ai/azure-ai-agents/samples/agents_tools/sample_agents_deep_research.py

```diff
@@ -32,131 +32,255 @@
        in the Management Center of your AI Foundry project.
 """
 
-import os, time
+import os
+import time
+import re
 from typing import Optional
 from azure.ai.projects import AIProjectClient
 from azure.identity import DefaultAzureCredential
 from azure.ai.agents import AgentsClient
 from azure.ai.agents.models import DeepResearchTool, MessageRole, ThreadMessage
 
 
+def convert_citations_to_superscript(markdown_content):
+    """
+    Convert citation markers in markdown content to HTML superscript format.
+
+    This function finds citation patterns like 【78:12†source】 and converts them to
+    HTML superscript tags <sup>12</sup> for better formatting in markdown documents.
+    It also consolidates consecutive citations by sorting and deduplicating them.
+
+    Args:
+        markdown_content (str): The markdown content containing citation markers
+
+    Returns:
+        str: The markdown content with citations converted to HTML superscript format
+    """
+    # Pattern to match 【number:number†source】
+    pattern = r"【\d+:(\d+)†source】"
```

Comment:
Let us avoid Unicode characters literal and compile a regex:
```python
pattern = re.compild("\u3010\d+:(\d+)†source\u3011")
```
then we can use shorter substitution:
```python
pattern.sub(replacement, markdown_content)
```